GithubApp/Controllers/FilterViewController.swift

```diff
@@ -9,27 +9,90 @@ import UIKit
 
 class FilterViewController: UIViewController {
 
+    //MARK: Instance Properties
+    var delegate: FilterDelegate?
+    var filterData :[Filter] = []
+    let checkedImage = UIImage(named: "check")! as UIImage
+    let uncheckedImage = UIImage(named: "uncheck")! as UIImage
+
+    // MARK: IB outlet
     @IBOutlet var filterView: FilterView!
+    
     override func viewDidLoad() {
-        super.viewDidLoad()
+        //super.viewDidLoad()
+        print("View Did load 2 called")
         filterView.filterTableView.delegate = self
         filterView.filterTableView.dataSource = self
     }
-    let languagesData = ["Python","Interior Painting","Electrician","Inappropriate client behavior","Gardener","Carpenter","House Cleaning"]
+    
+    // MARK: Functions
+    func configure(_ filterData: [Filter]){
+        self.filterData.removeAll()
+        self.filterData = filterData
+    }
+
+    // MARK: IB Actions
+    @IBAction func clearFilter(_ sender: Any) {
+        for index in 0..<filterData.count{
+            filterData[index].isSelected = false
+        }
+        delegate?.getSelectedFilters(filters: filterData)
+        dismiss(animated: true)
+    }
+    
+    @IBAction func applyFilter(_ sender: Any) {
+        delegate?.getSelectedFilters(filters: filterData)
+        dismiss(animated: true)
+    }
+
+    @IBAction func closeModal(_ sender: Any) {
+        dismiss(animated: true)
+    }
+
+    @IBAction func radioButtonAction(_ sender: UIButton) {
+        let tagNo: Int = sender.tag
+        if filterData[tagNo].isSelected == false {
```

Comment:
isSelected is already a boolean. It is good practice to use booleans directly in your conditions like
```
if filterData[index].isSelected { ...
```
OR 
```
if !filterData[index].isSelected { ...
```

Reply:
Done